File: ingest-ideas.mjs
import path from 'path';
import fs from 'fs';

// We need to use the vector-store.ts logic directly
// The issue: Chroma v2 uses ULIDs internally but the REST API expects UUIDs for collection paths
// However, the actual collection UUID from Chroma is an 8-4-4-4-12 format (looks like UUID but isn't valid UUIDv4)
// 
// Solution: use the Chroma Python client or call through the Node module that already works
// Since auto-ingest.ts works (it reported "Ingested: 4"), let's use the same approach

const { ChromaClient } = await import('chroma-js');

const content = fs.readFileSync('C:/Users/bryan/.openclaw/workspace/reports/income/high-ticket-automation-ideas.md', 'utf8');

const client = new ChromaClient({ host: 'http://localhost:8000' });

try {
  // Get the collection
  const collection = await client.getCollection({ name: 'agent_memory' });
  console.log('Collection found:', collection.name, collection.id);
  
  // Embed via Ollama
  const OllamaUrl = 'http://localhost:11434';
  const embRes = await fetch(`${OllamaUrl}/api/embeddings`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ model: 'nomic-embed-text', prompt: content.slice(0, 8000) })
  });
  const { embedding } = await embRes.json();
  console.log('Embedding done, size:', embedding.length);
  
  // Add to Chroma using the collection method
  await collection.add({
    ids: ['income-high-ticket-automation-ideas-2026-03-27'],
    documents: [content],
    embeddings: [embedding],
    metadatas: [{ source: 'report', type: 'income_ideas', date: '2026-03-27' }]
  });
  
  console.log('SUCCESS: ingested to agent_memory');
} catch (e) {
  console.error('Error:', e.message);
  // Try alternative: use raw REST with name path
  console.log('Trying raw REST with collection name...');
}

File: ingest-ideas.py
import requests
import json

# Embed content
content = open('C:/Users/bryan/.openclaw/workspace/reports/income/high-ticket-automation-ideas.md', 'r').read()

r = requests.post('http://localhost:11434/api/embeddings', json={'model': 'nomic-embed-text', 'prompt': content[:8000]})
embedding = r.json()['embedding']

col_id = '30b90900-28a6-4e72-a661-4ce733e8bf44'
doc_id = 'income-high-ticket-automation-ideas-2026-03-27'
payload = {
    'ids': [doc_id],
    'embeddings': [embedding],
    'documents': [content],
    'metadatas': [{'source': 'report', 'type': 'income_ideas', 'date': '2026-03-27'}]
}

# Try api/v2 endpoint
r2 = requests.post(f'http://localhost:8000/api/v2/collections/{col_id}/add', json=payload)
print(f'api/v2 add: {r2.status_code} - {r2.text[:200]}')

# Also try the older endpoint style
payload2 = {'ids': [doc_id], 'documents': [content], 'embeddings': [embedding]}
r3 = requests.post(f'http://localhost:8000/collections/{col_id}/add', json=payload2)
print(f'legacy add: {r3.status_code} - {r3.text[:200]}')
